src/flows/shared/steps/OrganizeCollection/Editor/StagingArea.tsx

```
@@ -72,63 +251,145 @@ function StagingArea({ tokensRef, stagedItems }: Props) {
+                  const columns = localStagedCollection[sectionId].columns;
+                  const size = IMAGE_SIZES[columns];
+                  const stagedItemRef = nftFragmentsKeyedByID[item.id];
+                  if (isEditModeToken(item) && stagedItemRef) {
+                    return (
+                      <SortableStagedNft
+                        key={item.id}
+                        tokenRef={stagedItemRef}
+                        size={size}
+                        mini={columns > 4}
+                      />
+                    );
+                  }
+                  return <SortableStagedWhitespace key={item.id} id={item.id} size={size} />;
+                })}
+              </SortableContext>
+            </DroppableSection>
+          ))}
         </SortableContext>
-        <DragOverlay adjustScale dropAnimation={defaultDropAnimationConfig}>
-          {activeItem && activeItemRef ? (
-            <StagedItemDragging
-              tokenRef={activeItemRef}
-              isEditModeToken={isEditModeToken(activeItem)}
-              size={itemWidth}
-            />
+        <DragOverlay dropAnimation={dropAnimation}>
+          {activeId ? (
+            sectionIds.includes(activeId as string) ? (
+              <SectionDragging
+                items={localStagedCollection[activeId].items}
+                itemWidth={IMAGE_SIZES[localStagedCollection[activeId].columns]}
+                columns={localStagedCollection[activeId].columns}
+                nftFragmentsKeyedByID={nftFragmentsKeyedByID}
+              />
+            ) : (
+              activeItem && (
+                <StagedItemDragging
+                  tokenRef={activeItemRef || null}
+                  isEditModeToken={isEditModeToken(activeItem)}
+                  size={
+                    IMAGE_SIZES[
+                      findSection(activeId)
+                        ? localStagedCollection[findSection(activeId)!].columns
```

nit: if you want to avoid using `findSection(activeId)!`, in the `findSection` func you can throw an error instead of returning `undefined`. that way the type signature will say it always returns a string

---

done!